src/main/java/dk/kb/datahandler/facade/DsDatahandlerFacade.java

```diff
@@ -94,57 +96,53 @@ public static ArrayList<String> ingestFromZipfile(String origin, InputStream is)
+        if (mTimeFrom==null) {
+            mTimeFrom=0L;
+        }
         
-        //Below two lines would work if client was generated with streaming Api
-        //DsPresentApi presentClient = getDsPresentApiClient();
-        //StreamingOutput so = presentClient.getRecords(oaiTarget, 0L, -1L, "SolrJSON");
-                
-        // TODO: Convert this to ds-present client when the client supports true streaming
-         URL presentUrl= new URIBuilder(ServiceConfig.getDsPresentUrl()+"/records")                           
-                            //.setPath("records") //This will set whole path, needs the /records above                           
-                           .setParameter("maxRecords", "100000")
-                           .setParameter("format", "SolrJSON")
-                           .setParameter("origin", origin)
-                           .build().toURL();
-                        
-         log.info("present URL protocol:"+presentUrl.getProtocol());
-        //String presentUrl =ServiceConfig.getDsPresentUrl()+"/records?&maxRecords=100000&format=SolrJSON&collection="+origin;
-                                
+        //Define the 2 clients. Inputstream from dsPresent will be feed direcly to the solrClient. 
+        DsPresentClient presentClient = new DsPresentClient(ServiceConfig.getDsPresentUrl());
         URL solrUpdateUrl=new URIBuilder(ServiceConfig.getSolrUrl())
-                          .setParameter("commit", "true")
-                          .build().toURL();      
-                                 
-        HttpURLConnection httpURLConnection = (HttpURLConnection) presentUrl.openConnection();        
-        httpURLConnection.setRequestMethod("GET");        
-        try (InputStream inputStream = httpURLConnection.getInputStream()){                 
-            //Give input stream to the POST request.        
-            HttpURLConnection solrServerConnection = (HttpURLConnection) solrUpdateUrl.openConnection();             
-            String solrResponse = HttpPostUtil.callPost(solrServerConnection, inputStream , "application/json");
-            log.info("Response from solr:"+solrResponse); //Example: {  "responseHeader":{    "rf":1,    "status":0,    "QTime":1348}}           
-
-            if (solrResponse.indexOf("\"status\":0") < 0) {
-                throw new IOException ("Unexpected status from solr:"+solrResponse);
-            }            
-        }
-        catch(Exception e) { //
-            log.error("Error calling ds-present or solr.",e);
-            throw new InternalServiceException("Error calling ds-present or solr");
-        }
-                      
+            .setParameter("commit", "true")
+            .build().toURL();      
      
+                
+        boolean hasMore=true;        
+        long batchSize=100L; //There seems to no performance boost increasing to 1000. Better keep in small batches.           
```

Comment:
Currently we have too few records to really measure and in the future we might ass speech-to-text and get quite large records. Please make this a property (with default as you choose).

Reply:
Added